day_10/task_2/pipe_maze.cpp

```diff
@@ -73,31 +74,31 @@ std::pair<MazePoint, MazePoint> AttributedMaze::find_next_neighbors(const MazePo
     const MazePoint current_position(maze_point);
     std::vector<MazePoint> neighbors;
 
-    if(is_connected_to_north(
+    if((current_position.first - 1)< maze_.size() && is_connected_to_north(
            maze_[current_position.first][current_position.second].tile) &&
        is_connected_to_south(
            maze_[to_north(current_position).first][current_position.second].tile))
     {
         neighbors.push_back(MazePoint(to_north(current_position)));
     }
 
-    if(is_connected_to_south(
+    if((current_position.first +1 )< maze_.size() && is_connected_to_south(
            maze_[current_position.first][current_position.second].tile) &&
        is_connected_to_north(
            maze_[to_south(current_position).first][current_position.second].tile))
     {
         neighbors.push_back(MazePoint(to_south(current_position)));
-    }
+    }   
 
-    if(is_connected_to_east(
+    if(current_position.second + 1 < maze_[0].size() && is_connected_to_east(
```

Comment:
```suggestion
    if(static_cast<std::size_t>(current_position.second + 1) < maze_[0].size() && is_connected_to_east(
```

Implicit conversion is bad smell

Reply:
done